### PR TITLE
Setting the targetFramework Dynamically

### PR DESCRIPTION
**Pull request by Anipik**
Currently the buildtools gives error if we try to use them with netcoreapp3.0 

```
/git/machinelearning/Tools/crossgen/crossgen.csproj : error NU1202: Package Microsoft.NETCore.App 3.0.0-preview-27324-5 is not compatible with netcoreapp2.0 (.NETCoreApp,Version=v2.0). Package Microsoft.NETCore.App 3.0.0-preview-27324-5 supports: netcoreapp3.0 (.NETCoreApp,Version=v3.0)
/git/machinelearning/Tools/crossgen/crossgen.csproj : error NU1202: Package Microsoft.NETCore.App 3.0.0-preview-27324-5 is not compatible with netcoreapp2.0 (.NETCoreApp,Version=v2.0) / linux-x64. Package Microsoft.NETCore.App 3.0.0-preview-27324-5 supports: netcoreapp3.0 (.NETCoreApp,Version=v3.0)
```

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks/PackageFiles/crossgen.sh [hunk 1]
```diff
@@ -1,6 +1,12 @@
...
 set -euo pipefail
 
+__TargetFramework=${3:-}
+
+if [ -z "$__TargetFramework" ]; then
+    __TargetFramework="netcoreapp2.0"
+fi
+
 # Restores crossgen and runs it on all tools components.
 usage()
```
Comment (eerhardt) [inline src/Microsoft.DotNet.Build.Tasks/PackageFiles/crossgen.sh]: Does the usage need to be updated?

File: src/Microsoft.DotNet.Build.Tasks/PackageFiles/crossgen.sh [hunk 2]
```diff
@@ -1,6 +1,12 @@
 #!/usr/bin/env bash
 set -euo pipefail
 
+__TargetFramework=${3:-}
```
Comment (eerhardt) [inline src/Microsoft.DotNet.Build.Tasks/PackageFiles/crossgen.sh]: Shouldn't parameter checking go with the rest of the parameter checking below on line 84?

**Comment by danmosemsft** [comment]
`__TargetFramework="netcoreapp2.0"` .. I would have expected 2.1

**Comment by Anipik** [comment]
>  __TargetFramework="netcoreapp2.0" .. I would have expected 2.1

I was not sure if all the repos which still use buildtools would work fine with changing the netcoreapp version from 2.0 to 2.1 (As current version being run is 2.0)

However for machine learning repo i use the default version as 2.1

**Comment by danmosemsft** [comment]
Ah - fair enough.

**Pull request merged**